client/client.go

```diff
@@ -55,47 +62,37 @@ func (client *Client) Proxies() (map[string]*Proxy, error) {
 	}
 	for _, proxy := range proxies {
 		proxy.client = client
+		proxy.created = true
 	}
 
 	return proxies, nil
 }
 
-// NewProxy instantiates a new proxy instance. Note Create() must be called on
-// it to create it. The Enabled field must be set to true, otherwise the Proxy
-// will not be enabled when created.
-func (client *Client) NewProxy(proxy *Proxy) *Proxy {
-	if proxy == nil {
-		proxy = &Proxy{}
+// Generates a new uncommitted proxy instance. In order to use the result, the
+// proxy fields will need to be set and have `Save()` called.
+func (client *Client) NewProxy() *Proxy {
+	return &Proxy{
+		client: client,
 	}
-
-	proxy.client = client
-	return proxy
 }
 
-// Create creates a new proxy.
-func (proxy *Proxy) Create() error {
-	request, err := json.Marshal(proxy)
-	if err != nil {
-		return err
+// CreateProxy instantiates a new proxy and starts listening on the specified address.
+// This is an alias for `NewProxy()` + `proxy.Save()`
```

Comment:
I left it exposed mostly because api_test.go uses it to create an already-disabled proxy, which is possible not really useful.
I'm not sure what you think about that use case, but I'd rather make it possible, rather than limiting things.